medusa/tv/episode.py

```diff
@@ -87,20 +84,209 @@
     get_scene_numbering,
     xem_refresh,
 )
-from medusa.tv.base import TV
-
-import shutil_custom
+from medusa.tv.base import Identifier, TV
 
 try:
     import xml.etree.cElementTree as ETree
 except ImportError:
     import xml.etree.ElementTree as ETree
 
-shutil.copyfile = shutil_custom.copyfile_custom
+logger = logging.getLogger(__name__)
 
-MILLIS_YEAR_1900 = datetime.datetime(year=1900, month=1, day=1).toordinal()
 
-logger = logging.getLogger(__name__)
+class InvalidNumbering(AssertionError):
+    """Numbering is invalid"""
+
+
+class EpisodeNumber(object):
+    """Base identifier class."""
+
+    _attrs = []
+
+    def __eq__(self, other):
+        """Compare episodes for equality."""
+        others = (
+            getattr(other, attr, NotImplemented)
+            for attr in self._attrs
+        )
+        return all(x==y for x,y in zip(self, others))
+
+    def __ne__(self, other):
+        """Check episodes for inequality."""
+        return not self == other
+
+    def __iter__(self):
+        for attr in self._attrs:
+            yield getattr(self, attr)
+
+    def __bool__(self):
+        """Check boolean value of episode numbering."""
+        return all(self)
```

Comment:
Sorry that was supposed to be `any(self)`.  But (with any) this would only return true if at least a single attribute was truthy (e.g. not zero, none, etc)